tests/regression_tests/deplete/test.py

```diff
@@ -130,3 +131,44 @@ def test_full(run_in_tmpdir, problem, multiproc):
 
     # Check that no additional tallies are loaded from the files
     assert np.all(n_tallies == 0)
+
+# Checks openmc.Materials objects can be created from depletion results
+def test_depletion_results_to_material(run_in_tmpdir, problem):
+
+    # Load the reference/test results
+    path_reference = Path(__file__).with_name('test_reference.h5')
+    res_ref = openmc.deplete.ResultsList.from_hdf5(path_reference)
+
+    # Firstly need to export materials.xml file for the initial simulation state
+    geometry, lower_left, upper_right = problem
+    materials = openmc.Materials()
+    for mat in geometry.root_universe.get_all_materials().values():
+        materials.append(mat)
+    materials.export_to_xml()
+
+    # Export last step of depletion to its own openmc.Materials object,
+    # using only nuclides available in the current nuclear data library
+    last_step_materials = res_ref.export_to_materials(-1)
+
+    # Check that the conversion of the final step depleted materials XML
+    # file hashes to what we expect.
+    output_xml_file = 'last_step_materials.xml'
+    last_step_materials.export_to_xml(path=output_xml_file)
+    with open(output_xml_file, 'rb') as result_file:
+        result_file_hash = hashlib.sha512()
+        for line in result_file:
+            result_file_hash.update(line)
```

Comment:
The hash based testing is nice because we don't have to track the potentially large xml file generated by this test. But it can hide the reason why this test is failing (#1338). I'd prefer checking if the strings are identical directly.  And (optionally) using [`difflib.unified_diff`](https://docs.python.org/3/library/difflib.html#difflib.unified_diff) to print the difference if two files differ

Reply:
Personally, I'd prefer not to make this change. Because this is a very minor piece of OpenMC, I would like to minimize the overall LOC we add to the repo in implementing this functionality.

Reply:
I'd prefer we not add more hashed results if at all possible. How big is the raw data we're talking about here?

Reply:
Not big, I am working on changing it to plaintext this morning.